pkg/medusa/hostmap.go

```diff
@@ -0,0 +1,206 @@
+}
+
+type backupGetter interface {
+	GetBackups(ctx context.Context) ([]*BackupSummary, error)
+}
+
+// getBackupRackIPs gets a map of racks to IPs from a Medusa CassandraBackup k8s object.
+func getSourceRacksIPs(k8sRestore medusaapi.CassandraRestore, client backupGetter, ctx context.Context) (map[NodeLocation][]HostDNSOrIP, error) {
+	backups, err := client.GetBackups(ctx)
+	if err != nil {
+		return nil, err
+	}
+	namedBackup, err := filterBackupsByName(k8sRestore.Spec.Backup, backups)
+	if err != nil {
+		return nil, err
+	}
+	out := make(map[NodeLocation][]HostDNSOrIP)
+	for _, i := range namedBackup.Nodes {
+		if i.Datacenter == k8sRestore.Spec.CassandraDatacenter.Name {
+			location := NodeLocation{
+				Rack: i.Rack,
+				DC:   i.Datacenter,
+			}
+			IP := HostDNSOrIP(i.Host)
+			if err != nil {
+				return nil, err
+			}
+			_, exists := out[location]
+			if exists {
+				out[location] = append(out[location], IP)
+			} else {
+				out[location] = []HostDNSOrIP{IP}
+			}
+		}
+	}
+	return out, nil
+}
+
+// getClusterRackFQDNs gets a map of racks to FQDNs from the current K8ssandraCluster k8s object. The CassDC does not exist yet, so we cannot refer to it for names.
+// We refer to the following code for how to calculate pod names: // https://github.com/k8ssandra/cass-operator/blob/master/pkg/reconciliation/construct_statefulset.go#L39
+func getTargetRackFQDNs(Kluster k8ssandraapi.K8ssandraCluster, dcName string) (map[NodeLocation][]HostName, error) {
+	cassDC, err := cassDCFromKluster(Kluster, dcName)
+	if err != nil {
+		return nil, err
+	}
+	racks := cassDC.GetRacks()
+	out := make(map[NodeLocation][]HostName)
+	for _, i := range racks {
+		location := NodeLocation{
+			DC:   cassDC.Name,
+			Rack: i.Name,
+		}
+		sizePerRack := int(cassDC.Spec.Size) / len(racks)
+		out[location] = getPodNames(Kluster.Name, cassDC.Name, i.Name, sizePerRack)
+	}
+	return out, nil
+}
+
+func getPodNames(clusterName string, DCName string, rackName string, rackSize int) []HostName {
+	out := []HostName{}
+	for i := 0; i < rackSize; i++ {
+		out = append(out, HostName(clusterName+"-"+DCName+"-"+rackName+"-sts-"+fmt.Sprint(i)))
+	}
+	return out
+}
+
+func cassDCFromKluster(Kluster k8ssandraapi.K8ssandraCluster, dcName string) (*cassdcapi.CassandraDatacenter, error) {
+	thisDC := k8ssandraapi.CassandraDatacenterTemplate{}
+	if len(Kluster.Spec.Cassandra.Datacenters) > 0 {
+		for _, i := range Kluster.Spec.Cassandra.Datacenters {
+			if i.Meta.Name == dcName {
+				thisDC = i
+			}
+		}
+	}
+	DCConfig := cassandrapkg.Coalesce(Kluster.Name, Kluster.Spec.Cassandra, &thisDC)
+	cassDC, err := cassandrapkg.NewDatacenter(
+		types.NamespacedName{
+			Namespace: Kluster.Namespace,
+			Name:      Kluster.Name,
+		},
+		DCConfig)
+	if err != nil {
+		return nil, err
+	}
+	return cassDC, nil
+}
+
+// GetHostMap gets the hostmap for a given CassandraBackup from IP sources to FQDN targets from the K8ssandraCluster and the backups returned by the Medusa gRPC client.
```

Comment:
As above, Godoc requirement is to repeat the function name. 

> Also, we shouldn't mention "IP" here either since we can get hostnames as sources.

That's valid, I'll change it.